tests/connection_test.py

```diff
@@ -33,6 +33,13 @@ def test_connect_tcp(request, create_connection, loop, server):
 
 
 @pytest.mark.run_loop
+def test_connect_tcp_timeout(request, create_connection, loop, server):
+    with pytest.raises(asyncio.TimeoutError):
+        yield from create_connection(
+            server.tcp_address, loop=loop, timeout=0)
```

Comment:
Zero timeout does not make much sense — cancelling connection on next loop iteration.
I think we should check if `timeout` is `None` or `>= 0`, if timeout is 0 replace it with `None`,
otherwise (if < 0) throw `ValueError`.